tests/posix/common/testcase.yaml

```diff
@@ -7,3 +7,6 @@ tests:
     # riscv toolchain lacks sys/timespec.h (maybe something else), so not
     # good for POSIX.
     arch_exclude: posix riscv32
+    # qemu_x86_64 currently uses Linux host toolchains, which leaks host
+    # headers into the build.
+    platform_exclude: qemu_x86_64
```

Comment:
Yeah, that's not what happening here.  What's actually happening is that the toolchain is **trying** to include a limits.h header from the "platform", and **not** finding one.  Our SDK compilers provide this header somehow (probably coming out of the sysroot they were built for), but a bare toolchain does not.

Technically this is a zephyr bug, I think.  The limits.h header is part of POSIX (as well as ISO C) and if we want to provide POSIX we need to have our own (or piggy back on the one from newlib where practical, I guess).